webui/react/src/hooks/useModal/UserSettings/useModalCreateUser.tsx

```diff
@@ -138,21 +197,29 @@ const ModalForm: React.FC<Props> = ({ form, user, groups, viewOnly }) => {
           </Select>
         </Form.Item>
       )}
-      {rbacEnabled && canModifyPermissions && !viewOnly && (
+      {rbacEnabled && user && canModifyPermissions && !viewOnly && (
         <Form.Item label={ROLE_LABEL} name={ROLE_NAME}>
           <Select mode="multiple" optionFilterProp="children" placeholder={'Add Roles'} showSearch>
-            {knownRoles.map((r) => (
-              <Select.Option key={r.id} value={r.id}>
-                {r.name}
-              </Select.Option>
-            ))}
+            {knownRoles
+              .filter(
+                (r) =>
+                  r.permissions.find((p) => p.isGlobal) &&
```

Comment:
From this page you should also be able to manage non-global roles for a user.

Reply:
Thinking on this, I think you should be able to _see_ all the roles a user has from this page but it's okay if you can't add or remove workspace specific roles. It's important to have a page where all of a user's roles can be seen.

Reply:
This page can see all the roles assigned to a user, however, for workspace specific roles, this page cannot see the associated workspace... For example, suppose an user has `workspaceAdmin` role scoped to workspace 1 and 2, this page can only see this user has `workspaceAdmin` role, and that's it